torch/nativert/executor/Placement.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <c10/core/Device.h>
+#include <c10/util/Logging.h>
+
+#include <optional>
+#include <unordered_map>
+
+namespace torch::nativert {
+
+/**
+ * @brief Normalizes a c10::Device by ensuring a canonical form.
+ *
+ * This function returns a normalized version of the input device:
+ * - For CPU devices, the returned device will have no index (i.e., the default CPU device).
+ * - For CUDA devices, if no index is specified, index 0 is assumed.
+ * - For other device types, the function will raise an error.
+ *
+ * @param device The input c10::Device to normalize.
+ * @return A normalized c10::Device with standardized indexing.
+ *
+ * @throws c10::Error If the device type is not CPU or CUDA.
+ */
+
+TORCH_API c10::Device normalizeDevice(const c10::Device& device);
+
+/**
+ * Returns true if the two devices are the same and has the same device index (if cuda).
+ */
+TORCH_API bool isSameDevice(
```

Comment:
Shouldn't be public either.

Reply:
The semantic feels so specific and different from what the name suggests that I don't think we want to expose this to all OSS users.

Reply:
i see. removed now